src/solvers/dg_tree/dg_2d.jl

```diff
@@ -231,38 +231,43 @@ end
 
   calcflux_twopoint_nonconservative!(f1, f2, u, element,
                                      have_nonconservative_terms(equations),
-                                     equations, dg, cache)
+                                     mesh, equations, dg, cache)
 end
 
 function calcflux_twopoint_nonconservative!(f1, f2, u::AbstractArray{<:Any,4}, element,
                                             nonconservative_terms::Val{false},
+                                            mesh::Union{TreeMesh{2}, CurvedMesh{2}, UnstructuredQuadMesh},
                                             equations, dg::DG, cache)
   return nothing
 end
 
 function calcflux_twopoint_nonconservative!(f1, f2, u::AbstractArray{<:Any,4}, element,
                                             nonconservative_terms::Val{true},
+                                            mesh::TreeMesh{2},
                                             equations, dg::DG, cache)
   #TODO: Create a unified interface, e.g. using non-symmetric two-point (extended) volume fluxes
   #      For now, just dispatch to an existing function for the IdealMhdEquations
   calcflux_twopoint_nonconservative!(f1, f2, u, element, equations, dg, cache)
 end
 
 
+# flux differencing volume integral. For curved meshes averaging of the
+# mapping terms, stored in `cache.elements.contravariant_vectors`, is peeled apart
+# from the evaluation of the physical fluxes in each Cartesian direction
 function calc_volume_integral!(du, u,
-                               mesh::TreeMesh{2},
+                               mesh::Union{TreeMesh{2}, CurvedMesh{2}, UnstructuredQuadMesh},
                                nonconservative_terms, equations,
                                volume_integral::VolumeIntegralFluxDifferencing,
                                dg::DGSEM, cache)
   @threaded for element in eachelement(dg, cache)
-    split_form_kernel!(du, u, nonconservative_terms, equations, volume_integral.volume_flux, dg, cache, element)
+    split_form_kernel!(du, u, nonconservative_terms, volume_integral.volume_flux, element,
+                       mesh, equations, dg, cache)
   end
 end
 
 @inline function split_form_kernel!(du::AbstractArray{<:Any,4}, u,
-                                    nonconservative_terms::Val{false}, equations,
-                                    volume_flux, dg::DGSEM, cache,
-                                    element, alpha=true)
+                                    nonconservative_terms::Val{false}, volume_flux, element,
+                                    mesh::TreeMesh{2}, equations, dg::DGSEM, cache, alpha=true)
```

Comment:
The `volume_flux` is a specification of the solver `dg`. Thus, we use the order `mesh, equations, volume_flux, dg, cache`, cf. https://trixi-framework.github.io/Trixi.jl/stable/styleguide/#Style-guide (comment applies to more places)

Reply:
Done. I updated the argument order for the `spilt_form_kernel` as well as `calcflux_twopoint`